orthopairs/src/main/java/org/reactome/release/orthopairs/OrthologyFileParser.java

```diff
@@ -0,0 +1,119 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class OrthologyFileParser {
+    private static final Logger logger = LogManager.getLogger();
+    private static Map<String,Map<String, Set<String>>> sourceTargetProteinHomologs = new HashMap<>();
+    private static Map<String,Map<String, Set<String>>> targetGeneProteinMap = new HashMap<>();
+
+    public static void parsePantherOrthologFiles(List<String> pantherFiles, String sourceMappingSpecies, JSONObject speciesJSONFile) throws IOException {
+
+        logger.info("Parsing homolog information from PANTHER files");
+        // Panther uses different naming conventions for species, which needs to be mapped to Reactome's 4-letter species keys
+        Set<String> pantherSpeciesNames = new HashSet<>();
+        String sourceSpeciesPantherName = "";
+        for (Object speciesKey : speciesJSONFile.keySet()) {
+            JSONObject speciesJSON = (JSONObject) speciesJSONFile.get(speciesKey);
+            if (!speciesKey.equals(sourceMappingSpecies)) {
+                pantherSpeciesNames.add(speciesJSON.get("panther_name").toString());
+            } else {
+                sourceSpeciesPantherName = speciesJSON.get("panther_name").toString();
+            }
+        }
+
+        if (!sourceSpeciesPantherName.equals("")) {
+            // There are 2 Panther files at time of writing since Sus Scrofa (PIG) info is found in a seperate file.
+            // The program iterates through both since for overlapping species it produces the same data structure. This is where the redundancy reduction of Sets comes in handy.
+            for (String pantherFileTar : pantherFiles) {
+                String extractedPantherFile = pantherFileTar.replace(".tar.gz", "");
```

Comment:
Since the pantherFile is already extracted when this executes, it may be better to pass in the `pantherFiles` list to this method already stripped of the .tar.gz extension.

Reply:
In Main.java, where you call `downloadAndExtractTarFile` you could move the loop into a method called `downloadAndExtractTarFiles(pantherFiles)` and have it return the new list of file names.

```
List<String> pantherFiles = downloadAndExtractTarFiles(Arrays.asList(pantherQfOFilename, pantherHCOPFilename));
```